File: Raven.Abstractions/Commands/DeleteCommandData.cs
//-----------------------------------------------------------------------
// <copyright file="DeleteCommandData.cs" company="Hibernating Rhinos LTD">
//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
using Raven.Abstractions.Data;
using Raven.Json.Linq;

namespace Raven.Abstractions.Commands
{
    /// <summary>
    /// A single batch operation for a document DELETE
    /// </summary>
    public class DeleteCommandData : ICommandData
    {
        /// <summary>
        /// Key of a document to delete.
        /// </summary>
        public virtual string Key { get; set; }

        /// <summary>
        /// Returns operation method. In this case DELETE.
        /// </summary>
        public string Method => "DELETE";

        /// <summary>
        /// Current document etag, used for concurrency checks (null to skip check)
        /// </summary>
        public virtual Etag Etag { get; set; }

        /// <summary>
        /// Information used to identify a transaction. Contains transaction Id and timeout.
        /// </summary>
        public TransactionInformation TransactionInformation { get; set; }

        /// <summary>
        /// RavenJObject representing document's metadata. In this case null.
        /// </summary>
        public RavenJObject Metadata
        {
            get { return null; }
        }

        /// <summary>
        /// Additional command data. For internal use only.
        /// </summary>
        public RavenJObject AdditionalData { get; set; }

        /// <summary>
        /// Translates this instance to a Json object.
        /// </summary>
        /// <returns>RavenJObject representing the command.</returns>
        public RavenJObject ToJson()
        {
            return new RavenJObject
                    {
                        {"Key", Key},
                        {"Etag", new RavenJValue(Etag != null ? (object) Etag.ToString() : null)},
                        {"Method", Method},
                        {"AdditionalData", AdditionalData}
                    };
        }
    }
}

File: Raven.Client.Java/lib/src/main/java/raven/abstractions/commands/DeleteCommandData.java
package raven.abstractions.commands;

import raven.abstractions.data.Etag;
import raven.abstractions.data.HttpMethods;
import raven.abstractions.data.TransactionInformation;
import raven.abstractions.json.linq.RavenJObject;
import raven.abstractions.json.linq.RavenJValue;

/**
 * A single batch operation for a document DELETE
 */
public class DeleteCommandData implements ICommandData {
  private String key;
  private Etag etag;
  private TransactionInformation transactionInformation;
  private RavenJObject additionalData;

  public DeleteCommandData(String key, Etag etag) {
    super();
    this.key = key;
    this.etag = etag;
  }

  public DeleteCommandData() {
    super();
  }

  @Override
  public RavenJObject getAdditionalData() {
    return additionalData;
  }

  @Override
  public Etag getEtag() {
    return etag;
  }

  @Override
  public String getKey() {
    return key;
  }

  @Override
  public RavenJObject getMetadata() {
    return null;
  }

  @Override
  public HttpMethods getMethod() {
    return HttpMethods.DELETE;
  }

  @Override
  public TransactionInformation getTransactionInformation() {
    return transactionInformation;
  }

  @Override
  public void setAdditionalData(RavenJObject additionalData) {
    this.additionalData = additionalData;
  }

  public void setEtag(Etag etag) {
    this.etag = etag;
  }

  public void setKey(String key) {
    this.key = key;
  }

  @Override
  public void setTransactionInformation(TransactionInformation transactionInformation) {
    this.transactionInformation = transactionInformation;
  }

  @Override
  public RavenJObject toJson() {
    RavenJObject object = new RavenJObject();
    object.add("Key", new RavenJValue(key));
    object.add("Etag", new RavenJValue(etag != null ? etag.toString() : null));
    object.add("Method", new RavenJValue(getMethod().name()));
    object.add("AdditionalData", additionalData);
    return object;
  }

}
